src/celeritas/io/RootFileManager.cc

```diff
@@ -35,5 +36,25 @@ void RootFileManager::write()
     CELER_ENSURE(write_status);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Create tree by providing its name and title.
+ *
+ * It is still possible to simply create a `TTree("name", "title")` in any
+ * scope that a RootFileManager exists, but this function explicitly shows the
+ * relationship between the newly created tree and `this->tfile_`.
+ *
+ * To expand this class to write multiple root files (one per thread), we just
+ * need to add a `tid` input parameter and call `tfile_[tid]->cd()`.
+ */
+detail::RootUniquePtr<TTree>
+RootFileManager::make_tree(const char* name, const char* title)
+{
+    CELER_EXPECT(tfile_->IsOpen());
+    detail::RootUniquePtr<TTree> uptree;
+    uptree.reset(new TTree(name, title, tfile_->cd()));
```

Comment:
```suggestion
    uptree.reset(new TTree(name, title, tfile_));
```
`TFile` is a `TDirectory` and `cd` has side-effect that you do not need.

Reply:
There is no constructor that accepts `TFile*`. This snippet, for example, does not compile:

```cpp
int main()
{
    auto tfile = new TFile("test.root", "recreate");
    auto ttree = new TTree("tree", "tree", tfile);
    tfile->Write();
    tfile->Close();

    return EXIT_SUCCESS;
}
```

Reply:
Yes, there is.  See my suggestion.  It takes 4 arguments:
```
uptree.reset(new TTree(name, title, 99, tfile_);
```